In [60]:
import numpy as np
import pandas as pd
from scipy.stats import ttest_rel, ttest_ind

df = pd.read_csv('adv_exp.txt')

df['idno'] = df['idno'].apply(int)

df.head()

cntry  idno  year  tvtot  ppltrst  pplfair  pplhlp  happy  sclmeet  sclact  \
0    CH   5.0     6    3.0      3.0     10.0     5.0    8.0      5.0     4.0   
1    CH  25.0     6    6.0      5.0      7.0     5.0    9.0      3.0     2.0   
2    CH  26.0     6    1.0      8.0      8.0     8.0    7.0      6.0     3.0   
3    CH  28.0     6    4.0      6.0      6.0     7.0   10.0      6.0     2.0   
4    CH  29.0     6    5.0      6.0      7.0     5.0    8.0      7.0     2.0   

   gndr  agea  partner  
0   2.0  60.0      1.0  
1   2.0  59.0      1.0  
2   1.0  24.0      2.0  
3   2.0  64.0      1.0  
4   2.0  55.0      1.0

In [3]:
df.cntry.unique()

array(['CH', 'CZ', 'DE', 'ES', 'NO', 'SE'], dtype=object)

1.) Did people become less trusting from 2012 to 2014? Compute results for each country in the sample.

In [56]:
from scipy.stats import ttest_rel

for cntry in df.cntry.unique():
    
    df12 = df[(df.cntry == cntry) & (df.year == 6) & (df.ppltrst.isna() == False)]
    df14 = df[(df.cntry == cntry) & (df.year == 7) & (df.ppltrst.isna() == False)]
    
    id_union = set(df12.idno).intersection(set(df14.idno))
    
    df12 = df12[[x in id_union for x in df12.idno]]
    df14 = df14[[x in id_union for x in df14.idno]]
    
    test = ttest_rel(df14['ppltrst'], df12['ppltrst'])
    
    print('{} paired t-test, test statistic: {}'.format(cntry, test[0]))
    print('{} paired t-test, p-value       : {}'.format(cntry, test[1]))
    print('')

CH paired t-test, test statistic: 0.6586851756725737
CH paired t-test, p-value       : 0.5102943511301135

CZ paired t-test, test statistic: 0.5001638336887216
CZ paired t-test, p-value       : 0.6171292682404208

DE paired t-test, test statistic: 0.18399501804849683
DE paired t-test, p-value       : 0.8568563797095805

ES paired t-test, test statistic: -2.4561906976601646
ES paired t-test, p-value       : 0.014181580725321856

NO paired t-test, test statistic: -0.5073077081124404
NO paired t-test, p-value       : 0.6120925701518767

SE paired t-test, test statistic: 2.0671082026033982
SE paired t-test, p-value       : 0.0390078167095776



Differences in people's assessments of others' trustworthiness changed to a statistically significant degree in two different countries. In Spain, people's assessment decreased, and in Sweden, it increased.

I used a paired t-test, because this compares answers among people surveyed during both years. Since this is a subjective measure, it's beneficial to control your sample to make sure it's the same people you are comparing. One person's "4" might be different from another's.

2.) Did people become happier?

In [57]:
for cntry in df.cntry.unique():
    
    df12 = df[(df.cntry == cntry) & (df.year == 6) & (df.happy.isna() == False)]
    df14 = df[(df.cntry == cntry) & (df.year == 7) & (df.happy.isna() == False)]
    
    id_union = set(df12.idno).intersection(set(df14.idno))
    
    df12 = df12[[x in id_union for x in df12.idno]]
    df14 = df14[[x in id_union for x in df14.idno]]
    
    test = ttest_rel(df14['happy'], df12['happy'])
    
    print('{} paired t-test, test statistic: {}'.format(cntry, test[0]))
    print('{} paired t-test, p-value       : {}'.format(cntry, test[1]))
    print('')

CH paired t-test, test statistic: 0.319412957862232
CH paired t-test, p-value       : 0.7495001355428359

CZ paired t-test, test statistic: 1.4561384833039597
CZ paired t-test, p-value       : 0.14584548433894534

DE paired t-test, test statistic: 0.8062257748298549
DE paired t-test, p-value       : 0.4346138707734991

ES paired t-test, test statistic: -1.613832417735418
ES paired t-test, p-value       : 0.1068245155648069

NO paired t-test, test statistic: -4.2856826576235925
NO paired t-test, p-value       : 2.0674530134057013e-05

SE paired t-test, test statistic: 0.5581637086030012
SE paired t-test, p-value       : 0.5768709591234161



People in Norway were substantially less happy in 2014. All other tests were not significant.

I check to make sure we have sufficient sample size, since that's a large drop.

In [59]:
df12 = df[(df.cntry == 'NO') & (df.year == 6) & (df.happy.isna() == False)]
df14 = df[(df.cntry == 'NO') & (df.year == 7) & (df.happy.isna() == False)]
    
id_union = set(df12.idno).intersection(set(df14.idno))
    
df12 = df12[[x in id_union for x in df12.idno]]
df14 = df14[[x in id_union for x in df14.idno]]

df14.shape

(727, 13)

727 is a sufficiently large sample.

I used a paired t-test, because this compares answers among people surveyed during both years. Since this is a subjective measure, it's beneficial to control your sample to make sure it's the same people you are comparing. One person's "4" might be different from another's.

3.) Who reported watching more TV in 2012, men or women?

In [77]:
print('Average number of hours of TV per week, men:   {}'.format(np.mean(df[(df.gndr == 1) & (df.year == 6)]['tvtot'])))
print('Average number of hours of TV per week, women: {}'.format(np.mean(df[(df.gndr == 2) & (df.year == 6)]['tvtot'])))

test = ttest_ind(df[(df.gndr == 1) & (df.year == 6)]['tvtot'].dropna(), df[(df.gndr == 2) & (df.year == 6)]['tvtot'].dropna())

print('T-test for statistical significant, t-score:   {}'.format(test[0]))
print('T-test for statistical significant, p-value:   {}'.format(test[1]))

Average number of hours of TV per week, men:   3.901906090190609
Average number of hours of TV per week, women: 3.944392523364486
T-test for statistical significant, t-score:   -0.6899928109209502
T-test for statistical significant, p-value:   0.4902360402696986


Women reported watching slightly more TV than men, but the difference was not statistically significant.

I used a regular t-test, since we were comparing the averages of two groups.

4.) Who was more likely to believe people were fair in 2012, people living with a partner or people living alone?

In [78]:
print('Average fairness rating, people living w/ partner:   {}'.format(np.mean(df[(df.partner == 1) & (df.year == 6)]['pplfair'])))
print('Average fairness rating, people living alone:        {}'.format(np.mean(df[(df.partner == 2) & (df.year == 6)]['pplfair'])))

test = ttest_ind(df[(df.partner == 1) & (df.year == 6)]['pplfair'].dropna(), df[(df.partner == 2) & (df.year == 6)]['pplfair'].dropna())

print('T-test for statistical significant, t-score:         {}'.format(test[0]))
print('T-test for statistical significant, p-value:         {}'.format(test[1]))

Average fairness rating, people living w/ partner:   6.080736012016523
Average fairness rating, people living alone:        5.856965174129353
T-test for statistical significant, t-score:         3.3201033970362084
T-test for statistical significant, p-value:         0.0009073952609883756


Those living with a partner were likely to rate others as more likely to be fair. The difference is statistically significant.

5.) Pick three or four of the countries in the sample and compare how often people met socially in 2014. Are there differences, and if so, which countries stand out?

In [88]:
df_2014 = df[df.year==7]

df_ch_2014 = df_2014[df_2014.cntry == 'CH']
df_cz_2014 = df_2014[df_2014.cntry == 'CZ']
df_de_2014 = df_2014[df_2014.cntry == 'DE']

print('Response, Switzerland: {}'.format(np.mean(df_ch_2014.sclmeet)))
print('Response, Cyprus:      {}'.format(np.mean(df_cz_2014.sclmeet)))
print('Response, Germany:     {}'.format(np.mean(df_de_2014.sclmeet)))

print()

print('T-test result, Switzerland and Cyrpus:')
print(ttest_ind(df_ch_2014.sclmeet.dropna(), df_cz_2014.sclmeet.dropna()))
print('T-test result, Cyprus and Germany:')
print(ttest_ind(df_cz_2014.sclmeet.dropna(), df_de_2014.sclmeet.dropna()))
print('T-test result, Switzerland and Germany:')
print(ttest_ind(df_ch_2014.sclmeet.dropna(), df_de_2014.sclmeet.dropna()))



Response, Switzerland: 5.160621761658031
Response, Cyprus:      4.445801526717557
Response, Germany:     4.428571428571429

Ttest_indResult(statistic=9.675850699914886, pvalue=1.7204743885392493e-21)
Ttest_indResult(statistic=0.0433661445611368, pvalue=0.9654226437207307)
Ttest_indResult(statistic=2.0861293215128325, pvalue=0.037289604714147245)


People from Switzerland reported meeting socially more often than people from Cyrpus and Germany. The difference was statistically significant in both cases.

6.) Pick three or four of the countries in the sample and compare how often people took part in social activities, relative to others their age, in 2014. Are there differences, and if so, which countries stand out?

In [89]:
df_ch_2014 = df_2014[df_2014.cntry == 'CH']
df_cz_2014 = df_2014[df_2014.cntry == 'CZ']
df_de_2014 = df_2014[df_2014.cntry == 'DE']

print('Response, Switzerland: {}'.format(np.mean(df_ch_2014.sclact)))
print('Response, Cyprus:      {}'.format(np.mean(df_cz_2014.sclact)))
print('Response, Germany:     {}'.format(np.mean(df_de_2014.sclact)))

print()

print('T-test result, Switzerland and Cyrpus:')
print(ttest_ind(df_ch_2014.sclact.dropna(), df_cz_2014.sclact.dropna()))
print('T-test result, Cyprus and Germany:')
print(ttest_ind(df_cz_2014.sclact.dropna(), df_de_2014.sclact.dropna()))
print('T-test result, Switzerland and Germany:')
print(ttest_ind(df_ch_2014.sclact.dropna(), df_de_2014.sclact.dropna()))


Response, Switzerland: 2.7816993464052286
Response, Cyprus:      2.703076923076923
Response, Germany:     2.7142857142857144

T-test result, Switzerland and Cyrpus:
Ttest_indResult(statistic=1.665721791943212, pvalue=0.09599050025440872)
T-test result, Cyprus and Germany:
Ttest_indResult(statistic=-0.04890685579084129, pvalue=0.9610082764822516)
T-test result, Switzerland and Germany:
Ttest_indResult(statistic=0.2746903478421981, pvalue=0.7836271662948312)


There were no statistically significant differences between the three countries for responses to this question.